### ISSUE TITLE
[BUG]: Compatibility issues with model loading and TensorBoard

### ISSUE THREAD

**Issue by jackeuylov** [MilesCranmer/PySR#779]
### What happened?

A bug happened!
When I am using model loading and TensorBoard simultaneously, I get the following error:
<img width="706" alt="888cde09aa6eb3d5025f862fbcffae9" src="https://github.com/user-attachments/assets/b6baf957-5381-42b8-90e5-068298850e5d" />
The case used is the code in Examples：
first, Using a model with TensorBoard to generate hall_of_fame：

```
import numpy as np
from pysr import PySRRegressor, TensorBoardLoggerSpec
rstate = np.random.RandomState(42)
X = rstate.uniform(-3, 3, (1000, 2))
y = np.exp(X[:, 0]) + X[:, 1]
logger_spec = TensorBoardLoggerSpec(
    log_dir="logs/run",
    log_interval=10,  # Log every 10 iterations
)
model = PySRRegressor(
    binary_operators=["+", "*", "-", "/"],
    logger_spec=logger_spec,
)
model.fit(X, y)

```
then, Using the model loading function to optimize the hall_of_fame.

```
import numpy as np
from pysr import PySRRegressor, TensorBoardLoggerSpec
rstate = np.random.RandomState(42)
X = rstate.uniform(-3, 3, (1000, 2))
y = np.exp(X[:, 0]) + X[:, 1]
model = PySRRegressor.from_file(run_directory="outputs/20241212_222454_uKch8W")
model.fit(X, y)
```

Then there will be the above error.

### Version

1.1.0

### Package Manager

None

### Interface

IPython Terminal

**Comment by MilesCranmer** [comment]
Thinking out loud, I think I know the issue. It’s the fact we don’t remove the logger from the state saved to the pickle file. Basically, we can’t pickle Julia objects. This new feature needs to be masked from the pickle

**Comment by MilesCranmer** [comment]
https://github.com/MilesCranmer/PySR/pull/780 should fix this
